src/orion/core/worker/consumer.py

```diff
@@ -100,6 +100,52 @@ def consume(self, trial):
             trial.status = 'broken'
             self.experiment.update_trial(trial, status=trial.status)
 
+    def get_execution_environment(self, trial, results_file='results.log'):
+        """Set a few environment variables to allow users and
+        underlying processes to know if they are running under orion
+
+        Parameters
+        ----------
+        results_file: str
+           file used to store results, this is only used by the legacy protocol
+
+        trial: Trial
+           reference to the trial object that is going to be run
+
+        Notes
+        ----
+        This functions define the environment variables described below
+
+        ORION_PROJECT: str
+           name of the experiment the user is currently working on.
+           A project is a set of experiments
+
+        ORION_EXPERIMENT: str
+           current experiment triplet that fully the experiment (i.e unique)
+
+        ORION_TRIAL_ID: str
+           current trial id that is currently being executed in this process
+
+        ORION_WORKING_DIRECTORY: str
+           orion current working directory
+
+        ORION_RESULTS_PATH: str
+           orion results file that is read by the legacy protocol to get the results of the trial
+           after a successful run
+
+        """
+        env = dict(os.environ)
+
+        env['ORION_PROJECT'] = self.experiment.name
+        env['ORION_EXPERIMENT'] = \
+            f'{self.experiment.name}-{self.experiment.version}-{self.experiment.metadata["user"]}'
```

Comment:
We support Python 3.5, so we cannot use `f-string` for formatting since they were introduced in Python 3.6